internal/crypto/container.go

```diff
@@ -25,9 +25,16 @@ func AuthenticateContainerRequestRFC6979(owner user.ID, pubBin, sig, payload []b
 	return nil
 }
 
-// AuthenticateContainerRequestN3 checks N3 witness of the container owner sent
+// AuthenticateContainerRequest checks N3 witness of the container owner sent
 // along with specified request payload against the current FS chain state.
-func AuthenticateContainerRequestN3(owner user.ID, invocScript, verifScript, payload []byte, fsChain N3ScriptRunner) error {
+//
+// If verifScript is a compressed ECDSA public key, AuthenticateContainerRequest
+// switches to deterministic ECDSA with SHA-256 hashing (RFC 6979) algorithm.
+func AuthenticateContainerRequest(owner user.ID, invocScript, verifScript, payload []byte, fsChain N3ScriptRunner) error {
+	if len(verifScript) == 33 && (verifScript[0] == 0x02 || verifScript[0] == 0x03) {
```

Comment:
PUSHINT32/PUSHINT64 in opcodes, so can technically be a valid script. Maybe it's better to decode the key (which has to be done anyway)? If it can be decoded --- it's a key, if not --- it's a script.